executor/importer/table_import.go

```diff
@@ -58,32 +58,45 @@ var (
 	CheckDiskQuotaInterval = time.Minute
 )
 
-func prepareSortDir(e *LoadDataController, taskID int64) (string, error) {
-	tidbCfg := tidb.GetGlobalConfig()
+// prepareSortDir creates a new directory for import, remove previous sort directory if exists.
+func prepareSortDir(e *LoadDataController, taskID int64, tidbCfg *tidb.Config) (string, error) {
 	sortPathSuffix := "import-" + strconv.Itoa(int(tidbCfg.Port))
-	sortPath := filepath.Join(tidbCfg.TempDir, sortPathSuffix, strconv.FormatInt(taskID, 10))
+	importDir := filepath.Join(tidbCfg.TempDir, sortPathSuffix)
+	sortDir := filepath.Join(importDir, strconv.FormatInt(taskID, 10))
 
-	if info, err := os.Stat(sortPath); err != nil {
+	info, err := os.Stat(importDir)
+	switch {
+	case err != nil && !os.IsNotExist(err):
```

Comment:
the switch-case is complicated, maybe if-else is better to read 😂

Reply:
done